src/ReportPortal.Shared/Reporter/TestReporter.cs

```diff
@@ -78,57 +81,60 @@ public void Start(StartTestItemRequest startTestItemRequest)
                     throw exp;
                 }
 
-                startTestItemRequest.LaunchUuid = LaunchReporter.Info.Uuid;
+                request.LaunchUuid = LaunchReporter.Info.Uuid;
                 if (ParentTestReporter == null)
                 {
-                    if (startTestItemRequest.StartTime < LaunchReporter.Info.StartTime)
+                    if (request.StartTime < LaunchReporter.Info.StartTime)
                     {
-                        startTestItemRequest.StartTime = LaunchReporter.Info.StartTime;
+                        request.StartTime = LaunchReporter.Info.StartTime;
                     }
 
-                    NotifyStarting(startTestItemRequest);
+                    NotifyStarting(request);
 
-                    var testModel = await _requestExecuter.ExecuteAsync(() => _service.TestItem.StartAsync(startTestItemRequest), null, LaunchReporter.StatisticsCounter.StartTestItemStatisticsCounter).ConfigureAwait(false);
+                    var testModel = await _requestExecuter.ExecuteAsync(() => _service.TestItem.StartAsync(request), null, LaunchReporter.StatisticsCounter.StartTestItemStatisticsCounter).ConfigureAwait(false);
 
                     _testInfo = new TestInfo
                     {
                         Uuid = testModel.Uuid,
-                        Name = startTestItemRequest.Name,
-                        StartTime = startTestItemRequest.StartTime
+                        Name = request.Name,
+                        StartTime = request.StartTime
                     };
 
                     NotifyStarted();
                 }
                 else
                 {
-                    if (startTestItemRequest.StartTime < ParentTestReporter.Info.StartTime)
+                    if (request.StartTime < ParentTestReporter.Info.StartTime)
```

Comment:
Can we move this logic to extension also, seems it is a part of normalization.

Reply:
Code from logs amender can't be moved to observer, cuz ILaunchReporter is not available from event handler.